File: CS/MVCDashboardDesigner/App_Start/DashboardConfig.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWeb;
using DevExpress.DashboardWeb.Mvc;
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using System.Web;
using System.Web.Routing;

namespace MVCDashboardDesigner.App_Start {
    public class DashboardConfig {
        public static void RegisterService(RouteCollection routes) {
            routes.MapDashboardRoute("api/dashboard", "DefaultDashboard");

            var dataBaseDashboardStorage = new DataBaseEditaleDashboardStorage(
                ConfigurationManager.ConnectionStrings["DashboardStorageConnection"].ConnectionString);

            DashboardConfigurator.Default.SetDashboardStorage(dataBaseDashboardStorage);

            DataSourceInMemoryStorage dataSourceStorage = new DataSourceInMemoryStorage();
            DashboardObjectDataSource objDataSource = new DashboardObjectDataSource("Object Data Source", typeof(SalesPersonData));
            objDataSource.DataMember = "GetSalesData";
            dataSourceStorage.RegisterDataSource("objectDataSource", objDataSource.SaveToXml());

            DashboardConfigurator.Default.SetDataSourceStorage(dataSourceStorage);

        }
    }
}

File: VB/MVCDashboardDesigner/App_Start/DashboardConfig.vb
Imports System.Configuration
Imports System.Web.Routing
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWeb
Imports DevExpress.DashboardWeb.Mvc
Imports MVCDashboardDesigner

Public Class DashboardConfig
    Public Shared Sub RegisterService(ByVal routes As RouteCollection)
        routes.MapDashboardRoute("api/dashboard", "DefaultDashboard")

        Dim dataBaseDashboardStorage = New DataBaseEditaleDashboardStorage(ConfigurationManager.ConnectionStrings("DashboardStorageConnection").ConnectionString)

        DashboardConfigurator.Default.SetDashboardStorage(dataBaseDashboardStorage)

        Dim dataSourceStorage As New DataSourceInMemoryStorage()
        Dim objDataSource As New DashboardObjectDataSource("Object Data Source", GetType(SalesPersonData))
        objDataSource.DataMember = "GetSalesData"
        dataSourceStorage.RegisterDataSource("objectDataSource", objDataSource.SaveToXml())

        DashboardConfigurator.Default.SetDataSourceStorage(dataSourceStorage)
    End Sub
End Class
